google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -943,23 +957,21 @@ private AsyncResultSet parseAndExecuteQueryAsync(
     Preconditions.checkNotNull(query);
     ConnectionPreconditions.checkState(!isClosed(), CLOSED_ERROR_MSG);
     ParsedStatement parsedStatement = getStatementParser().parse(query, this.queryOptions);
-    if (parsedStatement.isQuery()) {
-      switch (parsedStatement.getType()) {
-        case CLIENT_SIDE:
-          return ResultSets.toAsyncResultSet(
-              parsedStatement
-                  .getClientSideStatement()
-                  .execute(connectionStatementExecutor, parsedStatement.getSqlWithoutComments())
-                  .getResultSet(),
-              spanner.getAsyncExecutorProvider(),
-              options);
-        case QUERY:
-          return internalExecuteQueryAsync(parsedStatement, analyzeMode, options);
-        case UPDATE:
-        case DDL:
-        case UNKNOWN:
-        default:
-      }
+    switch (parsedStatement.getType()) {
+      case CLIENT_SIDE:
+        return ResultSets.toAsyncResultSet(
+            parsedStatement
+                .getClientSideStatement()
+                .execute(connectionStatementExecutor, parsedStatement.getSqlWithoutComments())
+                .getResultSet(),
+            spanner.getAsyncExecutorProvider(),
+            options);
+      case QUERY:
+      case UPDATE:
+        return internalExecuteQueryAsync(parsedStatement, analyzeMode, options);
```

Comment:
I think this also needs checks to verify that the result of the statement actually contains results and not just an update count.

Reply:
Done.